indico/modules/events/management/client/js/EmailAuthors.jsx

```diff
@@ -0,0 +1,211 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2022 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import abstractsPersonListURL from 'indico-url:abstracts.person_list';
+import contribsPersonListURL from 'indico-url:contributions.person_list';
+import sessionsPersonListURL from 'indico-url:sessions.person_list';
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import {Button, Dimmer, Loader, Message, Modal} from 'semantic-ui-react';
+
+import {EmailButton} from 'indico/modules/events/persons/Email';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Translate} from 'indico/react/i18n';
+
+import PersonList from './PersonList';
+
+// In abstracts there is always a submitter so we don't need an emptyListMessage
+const CONTEXTS = {
+  abstract: {personListURL: abstractsPersonListURL},
+  contribution: {
+    personListURL: contribsPersonListURL,
+    emptyListMessage: Translate.string(
+      'There are no persons associated with the selected contributions'
+    ),
+  },
+  session: {
+    personListURL: sessionsPersonListURL,
+    emptyListMessage: Translate.string(
+      'There are no persons associated with contributions in the selected sessions'
+    ),
+  },
+};
+
+const FILTER_OPTIONS = {
+  submitter: {
+    text: Translate.string('Submitters'),
+    isMatch: person => person.submitter,
+    contexts: ['abstract'],
+  },
+  speaker: {
+    text: Translate.string('Speakers'),
+    isMatch: person => person.speaker,
+  },
+  primaryAuthor: {
+    text: Translate.string('Primary authors'),
+    isMatch: person => person.primaryAuthor,
+  },
+  secondaryAuthor: {
+    text: Translate.string('Co-authors'),
+    isMatch: person => person.secondaryAuthor,
+  },
+  notRegistered: {
+    text: Translate.string('Not registered'),
+    isMatch: person => !person.registered,
+  },
+};
+
+const getIds = selector =>
```

Comment:
Since this function appears in more than one place now, should it be turned into a util? If so, where should I put it?

Reply:
I'll have to look around, for now I think duplicating it isn't a big deal since it's pretty small. If I find a good place to put it (and a more descriptive name) I'll let you know